src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
+        memcpy(&formattedIpv6Prefix[1], pMeshLocalPrefix->m8, OT_MESH_LOCAL_PREFIX_SIZE);
+
+        chip::Encoding::BytesToUppercaseHexString(formattedIpv6Prefix, OT_MESH_LOCAL_PREFIX_SIZE + 1,
+                                                  reinterpret_cast<char *>(*buffer), ReadLength);
+    }
+    break;
+
+    case ZCL_DIAG_OVERRUN_COUNT_ATTRIBUTE_ID: {
+        // TO DO
+        type = kTLVType_UnsignedInteger;
+        err  = CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+    break;
+
+    case ZCL_NEIGHBOR_TABLE_ATTRIBUTE_ID: {
+        type = kTLVType_List;
+        // List not yet functionnal
+        err = CHIP_ERROR_NOT_IMPLEMENTED;
+        // TO DO When list is functionnal.
+        // Determined limit of otNeighborInfo list
+        // ReadLength = sizeof(otNeighborInfo) * 20;
+        // *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        // VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        // otNeighborInfoIterator iterator = OT_NEIGHBOR_INFO_ITERATOR_INIT;
+        // otNeighborInfo neighInfo;
+
+        // uint16_t remainingSize = ReadLength;
+        // uint16_t offset        = 0;
+        // while (otThreadGetNextNeighborInfo(mOTInst, &iterator, &neighInfo) == OT_ERROR_NONE)
+        // {
+        //     if (remainingSize < sizeof(otNeighborInfo))
+        //     {
+        //         break;
+        //     }
+
+        //     memcpy(*buffer + offset, &neighInfo, sizeof(otNeighborInfo));
+        //     remainingSize -= sizeof(otNeighborInfo);
+        //     offset += sizeof(otNeighborInfo);
+        // }
+    }
+    break;
+
+    case ZCL_ROUTE_TABLE_ATTRIBUTE_ID: {
+        type = kTLVType_List;
+        // List not yet functionnal
+        err = CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+    break;
+
+    case ZCL_PARTITION_ID_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint32_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        Encoding::LittleEndian::Put32(*buffer, otThreadGetPartitionId(mOTInst));
+    }
+    break;
+
+    case ZCL_WEIGHTING_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint8_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        Encoding::Put8(*buffer, otThreadGetLeaderWeight(mOTInst));
+    }
+    break;
+
+    case ZCL_DATA_VERSION_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint8_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        Encoding::Put8(*buffer, otNetDataGetVersion(mOTInst));
```

Comment:
So... there's a lot of code duplication here.  Would it make sense to have a helper for "put this uint8_t value in", with the helper setting the type, allocating, etc; everything except the actual value-getting.